internal/charmstore/store.go

```diff
@@ -523,41 +512,43 @@ func (s *Store) UpdateBaseEntity(url *charm.Reference, update interface{}) error
 // This will be remedied when a new charm is uploaded by the promulgated
 // user. As promulgation is a rare operation, it is considered that the
 // chances this will happen are slim.
-func (s *Store) SetPromulgated(url *charm.Reference, promulgate bool) error {
-	entity, err := s.FindBestEntity(url)
```

Comment:
I wonder about the reasoning of the previous version making use of FindBestEntity